packages/vite-plugin-cloudflare/src/node-js-compat.ts

```diff
@@ -75,29 +75,41 @@ export function injectGlobalCode(
  */
 export function getNodeCompatAliases() {
 	return Object.fromEntries(
-		Object.entries(preset.alias)
-			.filter(([, to]) => !preset.external.includes(to))
-			.map(([from]) => [from, CLOUDFLARE_VIRTUAL_PREFIX + from]),
+		Object.keys(preset.alias).map((from) => [
+			from,
+			CLOUDFLARE_VIRTUAL_PREFIX + from,
+		]),
 	);
 }
 
 /**
  * Convert any virtual module Id that was generated by the aliases returned from `getNodeCompatAliases()`
- * back to real a module Id.
+ * back to real a module Id and whether it is an external (built-in) package or not.
  */
 export function resolveNodeAliases(
 	source: string,
 	workerOptions: SourcelessWorkerOptions,
 ) {
-	if (source.startsWith(CLOUDFLARE_VIRTUAL_PREFIX)) {
-		// We found a potential Node.js import.
-		// If Node.js compatibility is turned on, use the alias to the appropriate unenv polyfill.
-		// Otherwise return it to the original non-aliased import.
-		const from = source.slice(CLOUDFLARE_VIRTUAL_PREFIX.length);
-		if (isNodeCompat(workerOptions)) {
-			return preset.alias[from];
-		} else {
-			return from;
-		}
+	if (
+		!source.startsWith(CLOUDFLARE_VIRTUAL_PREFIX) ||
+		!isNodeCompat(workerOptions)
+	) {
+		return;
+	}
+
+	const from = source.slice(CLOUDFLARE_VIRTUAL_PREFIX.length);
+	const alias = preset.alias[from];
+	if (alias) {
+		return {
+			id: alias,
+			external: preset.external.includes(alias),
+		};
 	}
 }
+
+/**
+ * Get an array of modules that should be considered external.
+ */
+export function getNodeCompatExterns(): string[] {
```

Comment:
Subjective but I'd prefer if we avoid shortening names where possible. This only saves two characters and for me adds cognitive overhead.

```suggestion
export function getNodeCompatExternals(): string[] {
```

Reply:
https://github.com/flarelabs-net/vite-plugin-cloudflare/pull/57/commits/69957d984b2b3ec46e539ecc7da9bb7e8ab29a59